lib/rucio/rse/protocols/protocol.py

```diff
@@ -199,6 +199,24 @@ def __ligo(scope, name, rse, rse_attrs, protocol_attrs):
         from ligo_rucio import lfn2pfn as ligo_lfn2pfn  # pylint: disable=import-error
         return ligo_lfn2pfn.ligo_lab(scope, name, None, None, None)
 
+    @staticmethod
+    def __xenon(scope, name, rse, rse_attrs, protocol_attrs):
+        """
+        Given a LFN, turn it into a two level sub-directory structure based on the scope
+        plus a third level based on the name
+        :param scope: Scope of the LFN.
+        :param name: File name of the LFN.
+        :param rse: RSE for PFN (ignored)
+        :param rse_attrs: RSE attributes for PFN (ignored)
+        :param protocol_attrs: RSE protocol attributes for PFN (ignored)
+        :returns: Path for use in the PFN generation.
+        """
+        del rse
+        del rse_attrs
+        del protocol_attrs
+
+        return '%s/%s/%s/%s' % (scope[0:7], scope[4:len(scope)], name.split('-')[0]+"-"+name.split('-')[1], name)
```

Comment:
Hi, looks good to me. Please don't forget to add yourself to the author list at the top of the file.

Reply:
Thanks, done

Reply:
github actions seemed to have a hiccup, i've restarted the unittests

Reply:
I restarted the tests again after fixing a couple of style warning (flake8 apparently is very meticolous). They were just a missing space before and after the sum operator and I removed a tab in an empty line. Now also the autotests with client_syntax option have been completed successfully. All checks have passed now.